itis.playground/Pages/homework.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,254 @@
+        var temporaryNumerator  = fraction.numerator
+        var temporaryDenominator = fraction.denominator
+        while temporaryDenominator != 0 {
+            let temporary = temporaryNumerator % temporaryDenominator
+            temporaryNumerator = temporaryDenominator
+            temporaryDenominator = temporary
+        }
+        return temporaryNumerator
+    }
+    
+    func fractionReduction(fraction: Fraction) ->  Fraction {
+        let temporaryGCD = gcd(fraction: fraction)
+        let temporaryNumerator = fraction.numerator / temporaryGCD
+        let temporaryDenominator  = fraction.denominator / temporaryGCD
+        let reductionFraction = Fraction(numerator: temporaryNumerator, denominator: temporaryDenominator)
+        return reductionFraction
+    }
+    
+    func multiplication(firstFraction: Fraction, secondFraction: Fraction) -> Fraction {
+        let temporaryResult = Fraction(numerator: firstFraction.numerator * secondFraction.numerator, denominator: firstFraction.denominator * secondFraction.denominator)
+        return fractionReduction(fraction: temporaryResult)
+    }
+    
+    func division(firstFraction: Fraction, secondFraction: Fraction) -> Fraction {
+        let temporaryResult = Fraction(numerator: firstFraction.numerator * secondFraction.denominator, denominator: firstFraction.denominator * secondFraction.numerator)
+        return fractionReduction(fraction: temporaryResult)
+    }
+    
+    func addition(firstFraction: Fraction, secondFraction: Fraction) -> Fraction {
+        let temporaryDenominator = firstFraction.denominator * secondFraction.denominator
+        let temporaryNumerator = (temporaryDenominator / firstFraction.denominator * firstFraction.numerator) + (temporaryDenominator / secondFraction.denominator * secondFraction.numerator)
+        let temporaryResult = Fraction(numerator: temporaryNumerator, denominator: temporaryDenominator)
+        return fractionReduction(fraction: temporaryResult)
+    }
+    
+    func subtraction(firstFraction: Fraction, secondFraction: Fraction) -> Fraction {
+        let temporaryDenominator = firstFraction.denominator * secondFraction.denominator
+        let temporaryNumerator = (temporaryDenominator / firstFraction.denominator * firstFraction.numerator) - (temporaryDenominator / secondFraction.denominator * secondFraction.numerator)
+        let temporaryResult = Fraction(numerator: temporaryNumerator, denominator: temporaryDenominator)
+        return fractionReduction(fraction: temporaryResult)
+    }
+}
+
+let firstFraction: Fraction = Fraction(numerator: 1, denominator: 2)
+
+let secondFraction: Fraction = Fraction(numerator: 3, denominator: 7)
+
+let manipulation = Manipulations()
+
+var manipulationResult: () = manipulation.addition(firstFraction: firstFraction, secondFraction: secondFraction).output()
+
+    manipulationResult = manipulation.subtraction(firstFraction: firstFraction, secondFraction: secondFraction).output()
```

Comment:
сбвает с толку такой отступ 
```suggestion
manipulationResult = manipulation.subtraction(firstFraction: firstFraction, secondFraction: secondFraction).output()
```